lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1135,85 +1054,49 @@ export class UserAgentApplication {
      */
     private handleAuthenticationResponse(hash: string): void {
         // retrieve the hash
-        if (hash == null) {
-            hash = window.location.hash;
-        }
-
-        let self = null;
-        let isPopup: boolean = false;
-        let isWindowOpenerMsal = false;
-
-        // Check if the current window opened the iFrame/popup
-        try {
-            isWindowOpenerMsal = window.opener && window.opener.msal && window.opener.msal !== window.msal;
-        } catch (err) {
-            // err = SecurityError: Blocked a frame with origin "[url]" from accessing a cross-origin frame.
-            isWindowOpenerMsal = false;
-        }
+        const locationHash = hash || window.location.hash;
 
-        // Set the self to the window that created the popup/iframe
-        if (isWindowOpenerMsal) {
-            self = window.opener.msal;
-            isPopup = true;
-        } else if (window.parent && window.parent.msal) {
-            self = window.parent.msal;
-        }
+        // Check if the current flow is popup or hidden iframe
+        const iframeWithHash = WindowUtils.getIframeWithHash(locationHash);
+        const popUpWithHash = WindowUtils.getPopUpWithHash(locationHash);
+        const isPopupOrIframe = !!(iframeWithHash || popUpWithHash);
 
         // if (window.parent !== window), by using self, window.parent becomes equal to window in getResponseState method specifically
-        const stateInfo = self.getResponseState(hash);
+        const stateInfo = this.getResponseState(locationHash);
 
         let tokenResponseCallback: (response: AuthResponse, error: AuthError) => void = null;
 
-        self.logger.info("Returned from redirect url");
+        this.logger.info("Returned from redirect url");
```

Comment:
May be change this log statement to "returns from server redirect ..." to not confuse with our redirect flows.

Reply:
I can fix that in another PR. don't want have too much scope creep.